docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,295 @@
+# Allowing OpenShift Groups to be Synced With Sources on an LDAP Server and Making it Extensible for Other External Sources
+
+___
+Note: in this proposal, if a reference is prefixed with "OpenShift" or is typeset in `monospace`, it refers to an OpenShift object or field. For instance, the following table enumerates what references to OpenShift resources and LDAP entries may look like:
+
+OpenShift Resource | LDAP Entry
+------------------ | ----------
+OpenShift Group    | LDAP group
+OpenShift `Group`  | group
+`Group`            |
+___
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. One can indicate that an OpenShift `Group` mirrors an external representation by setting the `openshift.io/externalRecordProvider` `Annotation` in the `Group` as well as adding a `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
```

Comment:
Admins will sync, not link.  An admin syncs the groups and we automatically create the reverse links as part of that process.